CMakeLists.txt

```diff
@@ -447,9 +447,9 @@ foreach(target devilution ${BIN_TARGET})
     target_compile_definitions(${target} PRIVATE USE_SDL1)
   else()
     target_link_libraries(${target} PRIVATE
+      SDL2::SDL2main

```

Comment:
Curious that you need SDL2main specified first. I just build it on a clean Ubuntu 18.04 and didn't get any errors. Also Amiga requires it to be last (uses SDL1) because it's statically linking.

Reply:
http://www.cplusplus.com/forum/beginner/110753/

researched this and did

grep -r SDLmain to find it was located in CMakeLists.txt

this fixed the compile error
https://github.com/diasurgical/devilutionX/issues/686#issuecomment-609111543

Reply:
> Curious that you need SDL2main specified first. I just build it on a clean Ubuntu 18.04 and didn't get any errors. Also Amiga requires it to be last (uses SDL1) because it's statically linking.
seems to have built amiga fine
https://github.com/diasurgical/devilutionX/pull/687/checks?check_run_id=561411594

> amigaos-m68k - Success

Reply:
Yeah as mentioned Amiga uses SDL1 and you didn't change the order of it. But I'm wondering what would happen if SDL2 was statically linked in the same way on a a plaform.